gtda/plotting/plot_point_clouds.py

```diff
@@ -0,0 +1,244 @@
+        color-code the data points.
+
+    names: dict or None, optional, default: ``None``
+        Dictionary translating each numeric label into a string representing
+        its name. Should be of the format {label[int] : name[str]}.
+        If provided, a legend will be added to the plot.
+
+    dimension : int or None, default: ``None``
+        Sets the dimension of the resulting plot. If ``None``, the dimension
+        will be chosen between 2 and 3 depending on the shape of `point_cloud`.
+
+    plotly_params : dict or None, optional, default: ``None``
+        Custom parameters to configure the plotly figure. Allowed keys are
+        ``"trace"`` and ``"layout"``, and the corresponding values should be
+        dictionaries containing keyword arguments as would be fed to the
+        :meth:`update_traces` and :meth:`update_layout` methods of
+        :class:`plotly.graph_objects.Figure`.
+
+    marker_size : float or None, optional, default: 5
+        Sets the size of the markers in the plot. Must be a positive number.
+
+    opacity : float or None, optional, default: 0.8
+        Sets the opacity of the markers in the plot. Must be a number between
+        0 and 1.
+
+    to_scale : bool or None, optional, default: False
+        Whether or not to use the same scale across all axes of the plot.
+
+    show_plot: bool or None, optional, default: True
+        Whether or not to display the plot.
+
+    Returns
+    -------
+    fig : :class:`plotly.graph_objects.Figure` object
+        Figure representing a point cloud in 2D or 3D.
+
+    """
+
+    # If no labels provided, just enumerate data points, and record
+    # if there were user-provided labels to use in `names`.
+    labels_were_provided = labels is not None
+    if not labels_were_provided:
+        labels = np.arange(point_cloud.shape[0])
+
+    validate_params({"labels": labels},
+                    {"labels": {"type": (np.ndarray,),
+                                "of": {"type": (np.number,)}}})
+    validate_params({"names": names},
+                    {"names": {"type": (dict, type(None))}})
+    validate_params({"dimension": dimension},
+                    {"dimension": {"type": (int, type(None)),
+                                   "in": [2, 3]}})
+    validate_params({"marker_size": marker_size},
+                    {"marker_size": {"type": (float, int),
+                                     "in": Int(0, np.inf, closed="neither")}})
+    validate_params({"opacity": opacity},
+                    {"opacity": {"type": (float, int),
+                                 "in": Int(0, 1, closed="right")}})
+    validate_params({"to_scale": to_scale},
+                    {"to_scale": {"type": (bool,)}})
+    validate_params({"display_plot": display_plot},
+                    {"display_plot": {"type": (bool,)}})
+
+    if dimension is None:
+        dimension = np.min((3, point_cloud.shape[1]))
+
+    if names is not None:
+        if not labels_were_provided:
```

Comment:
do you really need the variable `labels_were_provided` ? Can you not do everything above at line 73?